migration-engine/core/src/commands/apply_migrations.rs

```diff
@@ -43,70 +44,57 @@ pub async fn apply_migrations(
+
+            match connector
+                .apply_script(unapplied_migration.migration_name(), &script)
+                .await
+            {
+                Ok(()) => {
+                    tracing::debug!("Successfully applied the script.");
+                    let p = connector.migration_persistence();
+                    p.record_successful_step(&migration_id).await?;
+                    p.record_migration_finished(&migration_id).await?;
+                    applied_migration_names.push(unapplied_migration.migration_name().to_owned());
+                    Ok(())
+                }
+                Err(err) => {
+                    tracing::debug!("Failed to apply the script.");
+
+                    let logs = err.to_string();
+
+                    connector
+                        .migration_persistence()
+                        .record_failed_step(&migration_id, &logs)
+                        .await?;
+
+                    Err(err)
+                }
+            }
+        };
+        fut.instrument(tracing::info_span!(
             "Applying migration",
             migration_name = unapplied_migration.migration_name(),
-        );
-        let _span = span.enter();
-        let migration_apply_start = Instant::now();
-
-        let script = unapplied_migration
-            .read_migration_script()
-            .map_err(ConnectorError::from)?;
-
-        tracing::info!(
-            script = script.as_str(),
-            "Applying `{}`",
-            unapplied_migration.migration_name()
-        );
-
-        let migration_id = connector
-            .migration_persistence()
-            .record_migration_started(unapplied_migration.migration_name(), &script)
-            .await?;
-
-        match connector
-            .apply_script(unapplied_migration.migration_name(), &script)
-            .await
-        {
-            Ok(()) => {
-                tracing::debug!("Successfully applied the script.");
-                let p = connector.migration_persistence();
-                p.record_successful_step(&migration_id).await?;
-                p.record_migration_finished(&migration_id).await?;
-                applied_migration_names.push(unapplied_migration.migration_name().to_owned());
-                let migration_duration_ms = Instant::now().duration_since(migration_apply_start).as_millis() as u64;
-                tracing::info!(
-                    migration_duration_ms = migration_duration_ms,
-                    "Migration executed in {}ms",
-                    migration_duration_ms
-                );
-            }
-            Err(err) => {
-                tracing::debug!("Failed to apply the script.");
-
-                let logs = err.to_string();
-
-                connector
-                    .migration_persistence()
-                    .record_failed_step(&migration_id, &logs)
-                    .await?;
-
-                return Err(err);
-            }
-        }
+        ))
+        .await?
```

Comment:
indentation change, pretty much. `span.enter()` does not interact well with async and the replacement requires one level of identation more